### PR TITLE
Problem: fix-unlucky-tx can't work with null indexer

### PR DESCRIPTION
**Pull request by yihuang**
Solution: support null tx indexer

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+- [#470](https://github.com/crypto-org-chain/cronos/pull/470) Run the tasks in `fix-unlucky-tx` command in parallel.
+- []() `fix-unlucky-tx` command supports `null` tendermint tx indexer.
```
Comment (yihuang) [inline CHANGELOG.md:8]:
```suggestion
- [#472](https://github.com/crypto-org-chain/cronos/pull/472) `fix-unlucky-tx` command supports `null` tendermint tx indexer.
```

File: cmd/cronosd/cmd/fix-unlucky-tx.go [hunk 2]
```diff
@@ -300,7 +301,7 @@ func newTxIndexer(config *tmcfg.Config, chainID string) (txindex.TxIndexer, erro
 		}
 		return es.TxIndexer(), nil
 	default:
-		return nil, fmt.Errorf("unsupported tx indexer backend %s", config.TxIndex.Indexer)
+		return &null.TxIndex{}, nil
```
Comment (yihuang) [inline cmd/cronosd/cmd/fix-unlucky-tx.go:304]: In this case, it only patches the block results, no tx indexer db at all.

File: cmd/cronosd/cmd/fix-unlucky-tx.go [hunk 3]
```diff
@@ -300,7 +301,7 @@ func newTxIndexer(config *tmcfg.Config, chainID string) (txindex.TxIndexer, erro
 		}
 		return es.TxIndexer(), nil
 	default:
```
Comment (thomas-nguy) [inline cmd/cronosd/cmd/fix-unlucky-tx.go:303]: maybe better to use case "null" instead of default (which is supposed to be kv)

**Review by thomas-nguy — APPROVED** [review]

File: cmd/cronosd/cmd/fix-unlucky-tx.go (hunk 3 above)
Comment (yihuang) [reply cmd/cronosd/cmd/fix-unlucky-tx.go:303]: https://github.com/tendermint/tendermint/blob/v0.34.x/node/node.go#L299
it match the tendermint's own logic though.

**Pull request merged**